milstoragepython/MilStoragePython.cpp

```diff
@@ -32,11 +32,27 @@ PYBIND11_PLUGIN(libmilstoragepython) {
     py::module m("libmilstoragepython", "Library to create, access and edit CoreML blob files.");
 
     py::class_<MilStoragePythonWriter> blobStorageWriter(m, "_BlobStorageWriter");
-    blobStorageWriter.def(py::init<const std::string&, bool>(), py::arg("file_name"), py::arg("truncate_file") = true)
-      .def("write_int8_data", &MilStoragePythonWriter::write_int8_data)
-      .def("write_uint8_data", &MilStoragePythonWriter::write_uint8_data)
-      .def("write_fp16_data", &MilStoragePythonWriter::write_fp16_data)
-      .def("write_float_data", &MilStoragePythonWriter::write_float_data);
+    blobStorageWriter.def(py::init<const std::string &, bool>(), py::arg("file_name"), py::arg("truncate_file") = true)
+        .def("write_int8_data", [](MilStoragePythonWriter &w, py::buffer buf) {
+            auto info = buf.request();
```

Comment:
@smpanaro - can you please explain this change a bit. This is preventing the data being copied? If so, how is it doing that.

Reply:
My understanding is that numpy arrays implement the [Python Buffer protocol](https://docs.python.org/3/c-api/buffer.html) which allows direct access to the underlying bytes. pybind11 has some [docs](https://github.com/pybind/pybind11/blob/ed466da571fbc1d711351eb818e4bf82adb99eca/docs/advanced/pycpp/numpy.rst) on this (which is where I found it), specifically the bit that starts with "To create a C++ function". I think "fast direct access" means "not copied". I also just noticed it's alluded to in the [readme](https://github.com/pybind/pybind11/tree/ed466da571fbc1d711351eb818e4bf82adb99eca#goodies) as well.

Re-visiting those numpy docs, I see a couple things that might be worth a look: some more rigorous safety checks on the inputs, maybe switching to `py::array`.